### PR TITLE
Issue19 floating joints

### PR DESCRIPTION
**Pull request by shaun-edwards**
Addresses issue #19 and part of #3.  This PR depends upon ros/robot_model#87.

<EDIT>
PR is against `hydro-devel` (I need it in hydro), let me know if you want me to issue a PR against `indigo-devel` as well.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
ping?

File: include/robot_state_publisher/joint_state_listener.h [hunk 1]
```diff
@@ -56,8 +56,11 @@ class JointStateListener{
 public:
   /** Constructor
    * \param tree The kinematic model of a robot, represented by a KDL Tree
+   * \param m Map of mimic joints
+   * \param ignore_joints List of joints to ignore when publishing transforms
    */
-  JointStateListener(const KDL::Tree& tree, const MimicMap& m);
+  JointStateListener(const KDL::Tree& tree, const MimicMap& m,
+                     const std::vector<std::string> ignore_joints);
```
Comment (isucan) [inline include/robot_state_publisher/joint_state_listener.h:63]: could this be a const &?

File: include/robot_state_publisher/robot_state_publisher.h [hunk 2]
```diff
@@ -77,7 +79,8 @@ class RobotStatePublisher
   void publishFixedTransforms(const std::string& tf_prefix);
 
 private:
-  void addChildren(const KDL::SegmentMap::const_iterator segment);
+  void addChildren(const KDL::SegmentMap::const_iterator segment,
+                   const std::vector<std::string>ignore_joints);
```
Comment (isucan) [inline include/robot_state_publisher/robot_state_publisher.h:83]: const &?

**Comment by isucan** [comment]
This is an ABI change too (for Hydro). However, this code is not widely liked against, as far as I know. I am fine with merging. @tfoote is this ok with you as well?

**Comment by shaun-edwards** [comment]
@isucan, thanks for the review.  I can make the desired changes.  Is there a way to make my changes ABI compatible?  If I include a wrapper method with the same signature as before?

If not, I'm OK just including this in Indigo.  This change is to support our calibration package, which is only a source release in Hydro.

**Comment by isucan** [comment]
That's a good point. Maybe we can keep the old constructor, and that is the same as an empty ignore_joints. Ping @tfoote @dirk-thomas

**Comment by tfoote** [comment]
Maintaining the old constructor will keep API compatibility, and should be done. ABI compatibility is hard to confirm without a checker. We're recompiling everything anyway at the moment so ABI compatibility is not strictly required. Also this is mostly used as an executable not a library.

**Comment by jackieokay** [comment]
ok, in order to close ros/robot_model#86 I'd like to close this issue and do something with this set of changes. Can I target it to Kinetic at least? I'll get an ABI checker to confirm everything is ok

**Comment by shaun-edwards** [comment]
My fault for not responding to @tfoote's request.  This issue may have been fixed in this #32 (already merged).  If so, I'm pretty sure this PR and it's changes are no longer needed.

**Comment by jackieokay** [comment]
no problem. I can't reproduce the error in Indigo, so I'll close this unless someone suggests otherwise.

**Pull request closed without merging**